src/Domain/Donation.php

```diff
@@ -357,6 +356,10 @@ public function __construct(
         ?string $tipAmount,
         ?RegularGivingMandate $mandate,
         ?DonationSequenceNumber $mandateSequenceNumber,
+        ?bool $giftAid = null,
```

Comment:
While the property is non-nullable I think it's clearer to make the param non-null too, and then we wouldn't need the false default set on line 395. Is there any advantage to having it default null here?

```suggestion
        bool $giftAid = false,
```

Reply:
Also not a blocker but I'm not sure if we should have a default at all, maybe better to not set a default here and pass the value from all callers to try and make sure we don't forget to consider it in any case we construct a donation. IntellJ can fill it in in all callers as part of the change signature refactoring command.

Reply:
In the more common case of the main donation form, we currently start off intentionally `null` on donation create. I think it's quite useful to distinguish between `false` and `null` there, and I'm not sure that we're less likely to miss a similar bug to this one in the main form if we were to switch to initialising it `false`. It kind of feels more risky than what we do now to me.

But then I'm casting to `false` anyway so maybe this is a symptom of having been awake too long, not sure.